core/agents/problem_solver.py

```diff
@@ -98,6 +99,7 @@ async def try_alternative_solutions(self) -> AgentResponse:
         self.next_state_iteration["alternative_solutions"][index]["tried"] = True
         self.next_state_iteration["description"] = llm_solution
         self.next_state_iteration["attempts"] = self.iteration["attempts"] + 1
+        self.next_state_iteration["status"] = IterationStatus.PROBLEM_SOLVER
```

Comment:
why is Problem Solver setting status `PROBLEM_SOLVER`? Can you double check this? It seems like Troubleshooter sets it and here it should be something else